packages/block-library/src/audio/edit.js

```diff
@@ -73,30 +67,6 @@ function AudioEdit( {
 		}
 	}, [] );
 
-	// We need to show the caption when changes come from
-	// history navigation(undo/redo).
-	useEffect( () => {
-		if ( caption && ! prevCaption ) {
-			setShowCaption( true );
-		}
-	}, [ caption, prevCaption ] );
-
-	// Focus the caption when we click to add one.
-	const captionRef = useCallback(
-		( node ) => {
-			if ( node && ! caption ) {
-				node.focus();
-			}
-		},
-		[ caption ]
-	);
-
-	useEffect( () => {
-		if ( ! isSelected && ! caption ) {
-			setShowCaption( false );
-		}
-	}, [ isSelected, caption ] );
```

Comment:
Note that I've also optimised these `useEffect` and `useCallback` calls. Previously these callbacks would get called whenever you type in the caption, now they're only called if emptiness changes.